spec/masscan_spec.rb

```diff
@@ -14,4 +14,53 @@
       expect(output_file.path).to eq(path)
     end
   end
+
+  describe '.scan' do
+    let(:ips) { '192.168.1.1' }
+
+    context 'when ip adresses are not given' do
+      it 'must raise an ArgumentError' do
+        expect {
+          subject.scan
+        }.to raise_error(ArgumentError, 'must specify at least one IP address')
+      end
+    end
+
+    context 'when masscan command was successful' do
+      let(:masscan_command) { Masscan::Command.new(ips: ips, output_file: tempfile.path) }
+      let(:tempfile)        { Tempfile.create(['masscan', '.json']) }
+
+      before do
+        allow(Masscan::Command).to receive(:new).and_return(masscan_command)
+        allow(Kernel).to receive(:system).with({}, 'masscan', '--output-filename', anything, ips).and_return(true)
+      end
+
+      it 'must return a Masscan::OutputFile' do
+        result = subject.scan(ips, ports: 80)
+
+        expect(result).to be_a(Masscan::OutputFile)
+        expect(result.path).to eq(masscan_command.output_file)
+      end
+    end
+
+    context 'when masscan command fails' do
+      before do
+        allow(Kernel).to receive(:system).with({}, 'masscan', '--output-filename', anything, ips).and_return(false)
```

Comment:
I think you can match certain arguments using a regex. That way we could at least test that the outputfile is within the `CACHE_DIR` path. Something like `%r{#{Regexp.escape(CACHE_DIR)}\/masscan-[^.]+\.json}`? Could also probably put that into a `let(:expected_output_filename) { ... }` variable at the top.